cmd/osbuild-pipeline/main.go

```diff
@@ -140,14 +140,15 @@ func main() {
 	}
 
 	repos := make([]rpmmd.RepoConfig, len(composeRequest.Repositories))
-	for i, repo := range composeRequest.Repositories {
+	for idx := range composeRequest.Repositories {
+		repo := composeRequest.Repositories[idx]
```

Comment:
I think your fix is correct, but the more idiomatic Go solution is "rebinding" in the loop instead of re-indexing the array.  Basically

```
for i, repo in composeRequest.Repositories {
    repo := repo
```

There's a lot more on this in https://go.dev/blog/loopvar-preview

Reply:
I personally find using the index more readable than making a copy / rebinding and I've used it all over the place, even when not strictly necessary.  It's also often useful when needing to index two aligned arrays (for copying or comparison), which admittedly doesn't apply here.

Great to see there's a change coming though.  I wasn't aware.  Thanks for the heads-up!

Reply:
Either way, since this is a backport of an existing change, there's no need to add extra work modifying the solution.